core/src/repair_generic_traversal.rs

```diff
@@ -187,7 +188,7 @@ pub fn get_closest_completion(
                 slot,
                 slot_meta,
                 limit - repairs.len(),
-                timestamp(),
+                now_timestamp,
```

Comment:
I see at least 3 loops inside this function + call to other functions.
Wouldn't this timestamp become stall depending of how much this code takes to run?

Reply:
Stacked repair timing generally shows ~100ms repair time over a 3s reporting period. There is a 100ms delay between iterations so the majority of the time should be spent sleeping. The timestamp is compared with tick granularity ~6ms/tick so still seems reasonable. Do you think this is worth updating to have have the timestamp passed as an `Option` for testing?

![Screenshot 2023-02-13 at 12 19 43 PM](https://user-images.githubusercontent.com/1054614/218569130-38f41753-1586-4196-81d4-b393a7b5c5d5.png)

Reply:
> Do you think this is worth updating to have have the timestamp passed as an Option for testing?

yeah, maybe.

Though in general not a fan of making api ambiguous for the sake of testing.
Maybe a few `sleep`s in the code is the least bad option in these scenarios.

Reply:
See #30345 which restores the `timestamp()` in `find_missing_indexes`.